app/controllers/payment-links/post-edit.controller.js

```diff
@@ -1,28 +1,26 @@
 'use strict'
 
-// NPM dependencies
 const lodash = require('lodash')
 
-// Local dependencies
-const logger = require('../../utils/logger')(__filename)
 const paths = require('../../paths')
 const productsClient = require('../../services/clients/products.client.js')
-const auth = require('../../services/auth.service.js')
-const { renderErrorView } = require('../../utils/response.js')
 
-module.exports = (req, res) => {
-  const gatewayAccountId = auth.getCurrentGatewayAccountId(req)
+module.exports = async function updatePaymentLink (req, res, next) {
+  const { productExternalId } = req.params
+  const gatewayAccountId = req.account.gateway_account_id
 
-  const editPaymentLinkData = lodash.get(req, 'session.editPaymentLinkData', {})
-  productsClient.product.update(gatewayAccountId, req.params.productExternalId, editPaymentLinkData)
-    .then(product => {
-      lodash.unset(req, 'session.editPaymentLinkData')
-      req.flash('generic', `Your payment link has been updated`)
-      res.redirect(paths.paymentLinks.manage)
-    })
-    .catch((err) => {
-      console.log(err)
-      logger.error(`[requestId=${req.correlationId}] update of payment link failed - ${err.message}`)
-      renderErrorView(req, res)
-    })
+  const editPaymentLinkData = lodash.get(req, 'session.editPaymentLinkData')
+  if (!editPaymentLinkData || editPaymentLinkData.externalId != productExternalId) {
+    req.flash('genericError', 'Something went wrong. Please try again.')
+    return res.redirect(paths.paymentLinks.manage)
+  }
+
+  try {
+    await productsClient.product.update(gatewayAccountId, productExternalId, editPaymentLinkData)
+    lodash.unset(req, 'session.editPaymentLinkData')
+    req.flash('generic', `Your payment link has been updated`)
+    res.redirect(paths.paymentLinks.manage)
+  } catch (err) {
+    return next(new Error(`Update of payment link failed. Error: ${err.message}`))
```

Comment:
Are we logging `correlationId` with this ?

Reply:
good point, I'll make it so it always logs it

Reply:
I've updated `express-unhandled-error-handler.js` in a new commit so that it logs the correlation ID.